### ISSUE TITLE
sticky components are desappearing

### ISSUE THREAD

**Issue by lfgtavora** [emilkowalski/vaul#191]
Hello! all `sticky `components are desappearing after open drawer, if i close or just drag a little they appears again. i already tryied everything nothing works. and i notice the same buggy behaviour in shadcn website :/

**Comment by dcodrin** [comment]
It seems to be related to the `overflow: hidden;` style, which gets appended to the `<html>` tag when the drawer opens. Interestingly, when removing the `<DrawerOverlay />` component from the `DrawerContent` component, it works even though the `overflow: hidden;`  style is still set. I hope this helps debug the issue.

**Comment by emilkowalski** [comment]
It's because the body gets position fixed and thus hides all sticky elements. This is unfortunately required in order for the drawer to work properly. I'm open to any suggestions here.

**Comment by dcodrin** [comment]
https://github.com/emilkowalski/vaul/assets/12925816/35a9bc33-d053-4283-8ddb-ac62b8ef694b

@emilkowalski Please check out the video. I simply removed `overflow:hidden` from the container and from the body and the sticky header is still there (sorry, I should have removed the overlay).

**Comment by emilkowalski** [comment]
@dcodrin That's true, but if we remove the overflow hidden in the actual code the drawer will break on mobile Safari.

**Comment by washedlih** [comment]
Also for some reason sticky components make the gesture animations very jittery on Safari and not as smooth.

**Comment by emilkowalski** [comment]
> Also for some reason sticky components make the gesture animations very jittery on mobile Safari and not as smooth.

Can you provide a codesandbox with reproduction?

**Comment by dcodrin** [comment]
@emilkowalski Unfortunately, as long as the overflows are added, sticky elements will not behave as expected. Not only this but other functionality, for example, using [useWindowScroll](https://github.com/streamich/react-use/blob/master/docs/useWindowScroll.md) which allows us to calculate the distance scrolled from the top, will reset back to 0 when the drawer is open.

You mentioned 
> the actual code the drawer will break on mobile Safari.

Perhaps we can find a workaround for this problem?

The drawer, btw, is amazing! And we would love to use it; it's very intuitive, but unfortunately, we cannot use it in production because of these issues.

**Comment by albingroen** [comment]
I'm running into the same issue...
https://github.com/emilkowalski/vaul/assets/19674362/f03a67e9-2ef7-478c-a8c0-6a3942091e46

**Comment by albingroen** [comment]
> This has been fixed in #209 

Sweet!
